libcxx/test/std/containers/sequences/vector.bool/sized_allocator.pass.cpp

```diff
@@ -0,0 +1,121 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// <vector>
+// vector<bool>
+
+// This test examines ambiguous calls to std::max in vector<bool>
+// Fix https://github.com/llvm/llvm-project/issues/121713
+
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <limits>
+#include <memory>
+#include <new>
+#include <vector>
+
+#include "test_macros.h"
+
+template <typename T, typename SIZE_TYPE = std::size_t, typename DIFF_TYPE = std::ptrdiff_t>
```

Comment:
```suggestion
template <typename T, typename Size = std::size_t, typename Difference = std::ptrdiff_t>
```

All caps template parameters are unusual in the codebase.

Reply:
Yeah, fixing the UB or ambiguous calls in my other PRs is a prerequisite for this one. Given that all these PRs and various tests utilize the `sized_allocator` class, I've elevated it to a support header file for the tests.